src/bgg_companion_api.py

```diff
@@ -14,21 +16,21 @@ def request(url: str):
 
 
 @cached(cache=TTLCache(maxsize=500, ttl=300))
-def get_users_collection(user: str) -> Union[None, dict]:
+def get_collection(user: str) -> Union[None, dict]:
     string_xml = request(f'https://boardgamegeek.com/xmlapi2/collection?username={user}')
     xml_parse = xmltodict.parse(string_xml)
     if "errors" in xml_parse and xml_parse["errors"]["error"]["message"] == "Invalid username specified":
-        return None
+        raise BoardGameIsNoneError("A board game was passed that does not exist within BoardGameGeek.")
     users_game_collection = xml_parse["items"]["item"]
     return users_game_collection
 
 
 @cached(cache=TTLCache(maxsize=500, ttl=300))
-def get_board_games(board_game_id: str) -> Union[None, list[BoardGame]]:
-    string_xml = request(f'https://api.geekdo.com/xmlapi2/thing?id={board_game_id}')
+def get_board_games(ids: tuple[str]) -> list[BoardGame]:
+    string_xml = request(f'https://api.geekdo.com/xmlapi2/thing?id={",".join(ids)}')
     xml_parse = xmltodict.parse(string_xml)
```

Comment:
Is this your own function?

Reply:
Nope, this is a library specifically for converting xml to a Python dictionary object.